model-optimizer/extensions/front/tf/CTCLossReplacement.py

```diff
@@ -78,109 +62,35 @@ def pattern(self):
+        merge_repeated_tf = ctc_greedy_decoder_tf.soft_get('merge_repeated', ctc_greedy_decoder_tf.id)
+        ctc_greedy_decoder = CTCGreedyDecoderSeqLenOp(graph, {'name': ctc_greedy_decoder_tf.name,
+                                                              'cmerge_repeated': merge_repeated_tf}).create_node()
+        ctc_greedy_decoder.in_port(0).connect(ctc_data_permute.out_port(0))
+        ctc_greedy_decoder.in_port(1).connect(ctc_greedy_decoder_tf.in_port(1).get_connection().get_source())
 
-        cast_fp_type = data_type_str_to_np(graph.graph['cmd_params'].data_type)
-        casted_seq_mask_node = Cast(graph, {'name': seq_len_tf_name + '/CastToFP32', 'dst_type': cast_fp_type}).create_node()
-        casted_seq_mask_node.in_port(0).connect(seq_mask_placeholder.out_port(0))
-        permuted_casted_seq_mask = create_op_with_const_inputs(graph, Transpose, {1: int64_array([1, 0])},
-                                                               {'name': seq_len_tf_name + '/Permute'})
-        permuted_casted_seq_mask.in_port(0).connect(casted_seq_mask_node.out_port(0))
-        rename_nodes([(seq_len_tf, seq_len_tf_name + '/AbandonedName'), (seq_mask_placeholder, seq_len_tf_name)])
-
-        # create CTCGreedyDecoder node and set mask node
-        ctc_merge_repeated_i = ctc_greedy_decoder_tf.soft_get('ctc_merge_repeated', ctc_greedy_decoder_tf.id)
-        ctc_greedy_decoder = CTCGreedyDecoderOp(graph, {'name': output_sparse_to_dense_name,
-                                                        'ctc_merge_repeated': ctc_merge_repeated_i}).create_node()
-        ctc_greedy_decoder.in_port(1).connect(permuted_casted_seq_mask.out_port(0))
-        rename_nodes([(sparse_to_dense_tf, output_sparse_to_dense_name + '/AbandonedName'),
-                      (ctc_greedy_decoder, output_sparse_to_dense_name)])
-
-        # create CTCLoss node and set attributes
-        assert ctc_loss_tf.has_valid('preprocess_collapse_repeated'), \
-            'The CTCLoss node "{}" misses "preprocess_collapse_repeated" attribute'.format(output_ctc_loss_name)
-        assert ctc_loss_tf.has_valid('ctc_merge_repeated'), \
-            'The CTCLoss node "{}" misses "ctc_merge_repeated" attribute'.format(output_ctc_loss_name)
-        assert ctc_loss_tf.has_valid('unique'), \
-            'The CTCLoss node "{}" misses "unique" attribute'.format(output_ctc_loss_name)
-        preprocess_collapse_repeated = ctc_loss_tf.preprocess_collapse_repeated
-        ctc_merge_repeated = ctc_loss_tf.ctc_merge_repeated
-        unique = ctc_loss_tf.unique
+        # set output of the new sub-graph as a source for SparseToDense consumer
+        output_ctc_loss_name = ctc_loss_tf.soft_get('name', ctc_loss_tf.id)
+        ctc_merge_repeated = ctc_loss_tf.soft_get('ctc_merge_repeated', ctc_loss_tf.id)
+        preprocess_collapse_repeated = ctc_loss_tf.soft_get('preprocess_collapse_repeated', ctc_loss_tf.id)
+        unique = ctc_loss_tf.soft_get('unique', ctc_loss_tf.id)
```

Comment:
this is incorrect since `unique` has boolean type. Now I am not sure that IR is correctly generated at all.
Please double check and inference as well